src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.ConfigUpdaters/AppSettingsConfigUpdater.cs

```diff
@@ -156,7 +156,10 @@ public Task<bool> IsApplicableAsync(IUpgradeContext context, ImmutableArray<Conf
             {
                 if (!jsonConfigFiles.Any(s => !string.IsNullOrEmpty(s.Configuration[setting.Key])))
                 {
-                    _appSettings.Add(setting.Key, setting.Value);
+                    if (!_appSettings.ContainsKey(setting.Key))
```

Comment:
This seems an ok precaution, but I wouldn't expect IConfigUpdater.IsApplicable to be called more than once in most cases. We may want to see why this is necessary and fix that also.

Reply:
This wasn't directly necessary, this was something I discovered as a precaution.

Reply:
you could just do _appSettings[setting.Key] = setting.Value;  here, it will do the same.

Reply:
i like that, simpler to read.